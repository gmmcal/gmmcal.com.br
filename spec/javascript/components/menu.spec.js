import { mount } from '@vue/test-utils'
import Menu from '@/components/menu'
import MenuItem from '@/components/menu_item'
import create from '../settings/utils'

describe('Menu', () => {
  let props, wrapper

  beforeAll(() => {
    props = {
      propsData: {
        items: create('menu_items')
      }
    }

    wrapper = mount(Menu, props)
  })

  test('Component is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('Component data properties', () => {
    const expected = ['active']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  test.skip('Scroll', () => {
  })

  describe('Child components', () => {
    test('MenuItem', () => {
      const child_component = wrapper.find(MenuItem)
      expect(child_component.isVueInstance()).toBeTruthy()
    })
  })
})
