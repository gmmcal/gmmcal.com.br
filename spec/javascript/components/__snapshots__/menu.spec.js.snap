// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<ul class="navbar-nav">
  <li class="active"><a data-turbolinks="false" href="#home" alt="home" class="home">Home</a></li>
  <li><a data-turbolinks="false" href="#about" alt="about" class="about">About</a></li>
</ul>
`;
